packages/parameters/src/appconfig/AppConfigProvider.ts

```diff
@@ -222,35 +248,43 @@ class AppConfigProvider extends BaseProvider {
    * For usage examples check {@link AppConfigProvider}.
    *
    * @param {string} name - The name of the configuration profile or its ID
-   * @param {GetAppConfigCombinedInterface} options - Options to configure the provider
+   * @param {AppConfigGetOptions} options - Options to configure the provider
    * @see https://docs.powertools.aws.dev/lambda-typescript/latest/utilities/parameters/
    */
-  public async get(
+  public async get<
+    ExplicitUserProvidedType = undefined,
+    InferredFromOptionsType extends
+      | AppConfigGetOptions
+      | undefined = AppConfigGetOptions
+  >(
     name: string,
-    options?: AppConfigGetOptionsInterface
-  ): Promise<undefined | string | Uint8Array | Record<string, unknown>> {
-    return super.get(name, options);
+    options?: InferredFromOptionsType & AppConfigGetOptions
+  ): Promise<
+    | AppConfigGetOutput<ExplicitUserProvidedType, InferredFromOptionsType>
+    | undefined
+  > {
+    return super.get(name, options) as Promise<
+      | AppConfigGetOutput<ExplicitUserProvidedType, InferredFromOptionsType>
+      | undefined
+    >;
   }
 
   /**
    * Retrieving multiple configurations is not supported by AWS AppConfig.
    */
-  public async getMultiple(
-    path: string,
-    _options?: unknown
-  ): Promise<undefined | Record<string, unknown>> {
-    return super.getMultiple(path);
+  public async getMultiple(path: string, _options?: unknown): Promise<void> {
```

Comment:
Should the return type here be `Promise<unknown>` ? 

Also, can't we provide the same option for the caller to pass explicit types to be expected, similar to `get`? I see, that it's not trivial to handle mixed return types, but getting list of the same type is possible. 

I see you did it for DynamoDBProvider.

Reply:
`void` is used when there's no `return` keyword. `unknown` would be a wider type in this case.

For `AppConfigProvider` specifically, the `getMultiple` operation is not supported at all and this method, when called, is guaranteed to throw (see `_getMultiple`). 

So since we are not returning anything, we use `void`.

Reply:
ok I wasn't aware of that, so it means user can call `getMultiple` and nothing happens and there will be no error or indication, except method doc. Seems like there is no way to prevent the method call 😕